File: app/src/main/java/expressive/fan/states/InitState.java
package expressive.fan.states;

import expressive.fan.core.MainActivityController;

public class InitState extends AbstractState {

    private MainActivityController controller;

    public  InitState(MainActivityController controller) {
        this.controller = controller;
        this.controller.setPlayButtonEnabled(true);
        this.controller.setPauseButtonEnabled(false);
        this.controller.setStopButtonEnabled(false);
    }

    @Override
    public void playClicked() {
        controller.createPlayer();
        controller.play();
        controller.setNextState(new PlayingState(controller));
    }

    @Override
    public void progressChanged(int progress) {
        controller.createPlayer();
        int total = controller.getTotalAudioDuration();
        int newPos = total / 100 * progress;
        controller.seekTo(newPos);
        controller.setNextState(new PausedState(controller));
    }
}

File: app/src/main/java/expressive/fan/states/InitState.kt
package expressive.fan.states

import expressive.fan.core.MainActivityController

class InitState(private val controller: MainActivityController) : AbstractState() {
    override fun playClicked() {
        controller.createPlayer()
        controller.play()
        controller.setNextState(PlayingState(controller))
    }

    override fun progressChangedByUser(progress: Int) {
        controller.createPlayer()
        controller.seekToNewPos(progress)
        controller.setNextState(PausedState(controller))
    }

    override fun progressChangedByScheduler() {
        controller.resetProgress()
    }

    init {
        controller.setPlayButtonEnabled(true)
        controller.setPauseButtonEnabled(false)
        controller.setStopButtonEnabled(false)
    }
}